framework/src/systems/NonlinearSystemBase.C

```diff
@@ -981,6 +981,12 @@ NonlinearSystemBase::setConstraintSecondaryValues(NumericVector<Number> & soluti
   std::map<std::pair<unsigned int, unsigned int>, PenetrationLocator *> * penetration_locators =
       NULL;
 
+  mooseAssert(displaced ? bool(_fe_problem.getDisplacedProblem().get()) : true,
+              "If we're calling this method with displaced = true, then we better well have a "
+              "displaced problem");
```

Comment:
```suggestion
  if (displaced)
    mooseAssert(_fe_problem.getDisplacedProblem().get(),
               "If we're calling this method with displaced = true, then we better well have a "
               "displaced problem");
```

With the assertion opt being:

```
if (displaced)
  (void)0
```

this is fine and will be removed by the compiler. I used to do this, but the ternary assertion with a true just looks like hot trash IMO